src/main/java/org/codehaus/mojo/flatten/FlattenMode.java

```diff
@@ -77,8 +77,12 @@
     /** Only resolves variables revision, sha1 and changelist. Keeps everything else. 
      * See <a href="https://maven.apache.org/maven-ci-friendly.html">Maven CI Friendly</a> for further details. 
      */
-    resolveCiFriendliesOnly;
+    resolveCiFriendliesOnly,
 
+    /**
+     * Fix for {@link #resolveCiFriendliesOnly}
+     */
+    version;

```

Comment:
So if this is a "fix" for `resolveCiFriendliesOnly` why do we have to introduce a new mode then?
My impression is also that this plugin can not be understood by an avergage user anymore. So what does `version` mode do? Where is it documentated? I already had concerns like this with the naming of `resolveCiFriendliesOnly` and the documentation is already lacking there but I did not want to be a blocker for urgently requested features. I think that your PR is in general providing a very helpful improvement as I also experienced some problems/bugs with this `resolveCiFriendliesOnly`. However the way we address this should IMHO need some improvement.
@lasselindqvist do you have some cents on this?

Reply:
I just kept it during rebase, no other reasons from my side. ^^